addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -635,22 +637,22 @@ function makeActionManager(env) {
         }
 
         const currentController = _getCurrentController();
-        let searchState;
-        if (currentController && currentController.getState) {
-            currentController.exportedState = currentController.getState();
-        }
-        if (currentController && currentController.getSearchState) {
-            searchState = currentController.getSearchState();
-            currentController.exportedSearchState = searchState;
+        if (currentController && currentController.getLocalState) {
+            currentController.exportedState = currentController.getLocalState();
         }
-
         if (controller.exportedState) {
             controller.props.state = controller.exportedState;
         }
-        if (options.shareSearchState && searchState) {
-            controller.props.searchState = searchState;
-        } else if (controller.exportedSearchState) {
-            controller.props.searchState = controller.exportedSearchState;
+
+        if (currentController && currentController.getGlobalState) {
```

Comment:
Remarks: 
- this thing seems useless for client actions.
- restore and swithView (at least) use this.
- if really globalState thing is not useful for client actions --> maybe use that thing in setupView instead of usesetupAction?